### PR TITLE
[ui] Show backfill policies, preview of backfill partition ranges

### PR DESCRIPTION
**Pull request by bengotow**
## Summary & Motivation

This PR adds:

- A new Backfill policy section to the asset sidebar and asset definition page, which displays the asset's backfill policy description, if a policy is present.

- A new Backfill Preview modal, which allows the user to see which ranges of each asset will be backfilled, and what backfill policy will be used for each asset. This modal is accessible via a Preview button in the Launch Partitions modal IF there are multiple backfill policies or if the partition mapping is in use.

- Smaller changes:
   + The Launch backfill button no longer says how many runs will be launched, since this number is not always known.
   + The partition selection modal no longer offers the "show more asset health bars" link which confusingly pasted asset health bars into the modal. This option is totally removed.
   + The "Backfill options" section is now just called "Options" (since we didn't say "Backfill partitions", etc.) and disappears entirely if there are no options to be displayed rather than appearing empty.
   
## How I Tested These Changes

I updated the existing tests, and wrote new tests verifying that the preview option is shown in the correct scenarios and opens the new modal. I also added new stories covering the backfill preview dialog contents.

New Preview button:

image

When partition mapping is present, the text changes to this:
image

New Preview panel:

image

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetLink.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export const AssetLink: React.FC<{
       style={{maxWidth: '100%'}}
     >
       {icon ? (
-        <Box margin={{right: 8, top: 2}}>
+        <Box margin={{right: 8, top: 2}} style={{width: 16}}>
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetLink.tsx:27]: "16px"

File: js_modules/dagster-ui/packages/ui-core/src/assets/BackfillPreviewModal.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    },
+  );
+
+  const partitionsByAssetToken = React.useMemo(() => {
+    return Object.fromEntries(
+      (data?.assetBackfillPreview || []).map((d) => [tokenForAssetKey(d.assetKey), d.partitions]),
+    );
+  }, [data]);
+
+  // BG Note: The transform: scale(1) below fixes a bug with MiddleTruncate where the text size
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/BackfillPreviewModal.tsx:68]: ah weird that change `scale` doesnt trigger the resize observer

Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/BackfillPreviewModal.tsx:68]: I know right! This is still blowing my mind a bit, will try to come up with a better / more global fix in a follow-up

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetLink.tsx (hunk 1 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetLink.tsx:27]: Ahh I think I can rm this, I was thinking it was causing the text truncation issue I added the note about, but turns out it was not 🙏

**Pull request merged**